datafusion/core/src/physical_plan/windows/bounded_window_agg_exec.rs

```diff
@@ -548,9 +548,9 @@ impl SortedPartitionByBoundedWindowStream {
             for (partition_row, WindowState { state: value, .. }) in window_agg_state {
                 let n_prune =
                     min(value.window_frame_range.start, value.last_calculated_index);
-                if let Some(state) = n_prune_each_partition.get_mut(partition_row) {
-                    if n_prune < *state {
-                        *state = n_prune;
+                if let Some(current) = n_prune_each_partition.get_mut(partition_row) {
```

Comment:
The changes in this file appear to be a drive by cleanup (as in not directly related to this PR) -- that is fine I am just confirming my understanding

Reply:
Right, this is a drive-by cleanup. There were too many `state`s going around making it hard to read 🙂